storage.js

```diff
@@ -99,12 +101,19 @@ class BigQueryStorage {
      * @return {Promise|null}
      */
     insert(rows) {
-        let data = Array.isArray(rows) === true ? rows.map(this.getInsertData) : this.getInsertData(rows);
+        let data = Array.isArray(rows) === true ? rows.map(this.mapRow) : this.mapRow(rows);
         return this.table.insert(data);
     }
+
+    mapRow(row) {
+        let insertData = this.getInsertData(row);
+        row._insertId = row._insertId || IdGenerator.generateInsertId(); // uuid
+        insertData.insertId = row._insertId;
+        return insertData;
```

Comment:
I think it would be better to just create the raw row here.
Something like this:
```suggestion
        let insertData = this.getInsertData(row);
        row._insertId = row._insertId || IdGenerator.generateInsertId();
        return { insertId: row._insertId, json: insertData }; // Or call some helper like Table.getRawRow(insertId, data);
```
This way, you avoid cloning objects down in the flow. Keep in mind that insertData is already a new object that the user doesn't have access to it.